packages/relay/src/lib/eth.ts

```diff
@@ -601,12 +605,22 @@ export class EthImpl implements Eth {
         this.logger.info(`${requestIdPrefix} Returning gas: ${response.result}`);
         return prepend0x(trimPrecedingZeros(response.result));
       } else {
+        this.logger.error(`${requestIdPrefix} No gas estimate returned from mirror-node: ${JSON.stringify(response)}`);
         return this.predefinedGasForTransaction(transaction, requestIdPrefix);
       }
     } catch (e: any) {
       this.logger.error(
         `${requestIdPrefix} Error raised while fetching estimateGas from mirror-node: ${JSON.stringify(e)}`,
       );
+      // in case of contract revert, we don't want to return a predefined gas but the actual error with the reason
+      if (
+        this.estimateGasThrows &&
```

Comment:
I understand this is a global setting, but should we honor it in this case? I mean, should we throw regardless of this setting in this case?

Reply:
It would be weird not to honour it,`estimateGasThrows` defaults to `true` if the environment variable is `undefined`, so I don’t see why we would want to bypass this if it’s going to be true in most cases. 

If the user has set this variable explicitly to `false` in his .env file, I believe we should also honour it here, what is your opinion?

Reply:
This is tricky, not sure. But if the corresponding `eth_call` fails, shouldn't the `eth_estimateGas` fail as well regardless of this setting?

Reply:
Hmm, it's kinda tricky because we rely on the fallback estimations for some tools to work properly right now 😆

Reply:
See https://github.com/hashgraph/hedera-json-rpc-relay/pull/2834#discussion_r1720163125